### DISCUSSION TITLE
Socket.io auth id

### DISCUSSION THREAD

**Discussion by Notadumdum** [miguelgrinberg/python-socketio#1114] [Q&A] [answered]
Good day 
![20230105_153427](https://user-images.githubusercontent.com/105162586/210804729-e72e9304-a7a2-4efe-bd9c-e2bfe87a2adf.jpg)

From what I know the sid is created and sent to the client upon every new connection
But
![20230105_153501](https://user-images.githubusercontent.com/105162586/210805030-827297af-fe15-4464-bbda-a1feda6b1a46.jpg)
The auth id remains.. even after any connection is closed or reopened... after a while it expires and a new one is created... the question I'm asking is.. If I'm to connect to a socket.io server using a non browser client, how can I get the server to recognize me as a ''valid user''... or can I use the already existing auth id...or can I connect to the server in a way that I'll be sent a new auth id and if so how would I be able to store that auth id

**Comment by miguelgrinberg** [comment, marked as answer]
Socket.IO does not specify how to do authentication, each server is free to implement this in any way it sees fit. You need to ask this question to the owner of the server you are connecting to.
